avocado/plugins/yaml_to_mux.py

```diff
@@ -433,6 +528,10 @@ def run(self, args):
                 entry.insert(0, '')  # add path='' (root)
             data.get_node(entry[0]).value[entry[1]] = entry[2]
 
+        mux_path = getattr(args, 'mux_path', None)
+        if mux_path is None:
+            mux_path = ['/run/*']
+
         data = apply_filters(data, getattr(args, 'mux_only', None),
                              getattr(args, 'mux_out', None))
-        args.mux.data_merge(data)
+        args.mux.add_variants_plugin(MuxPlugin(data, mux_path))
```

Comment:
This change defines a variant generation interface, which is great and looks appropriate.  What it doesn't, but we should try to, is to declare its interface (as in `avocado.core.plugin_interfaces`), and follow the "new style plugin" activation.

Something else to think about: it may be a nice time to properly state that what a Job has, that is, a `self.mux`, instead of `self.args.mux`. It may even deserve a rename, if you think that would make the purpose and borders clearer.